### PR TITLE
Improve Mayan integration

### PR DESCRIPTION
**Pull request by mohandesjiri**
- use onReady callback instead of onLoad
- toggle for bridge in/out
- add to navbar
- add referrer address
- set RPCs
- limit token list to the supported marginfi tokens
- support wallet connect v2

### CODE REVIEW

File: apps/marginfi-v2-ui/src/pages/bridge.tsx [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+            setSwapInitiateListener(callback: (swap: MayanSwapInfo) => void): void;
+            removeSwapInitiateListener(): void;
+            setSwapCompleteListener(callback: (swap: MayanSwapInfo) => void): void;
+            removeSwapCompleteListener(): void;
+            setSwapRefundListener(callback: (swap: MayanSwapInfo) => void): void;
+            removeSwapRefundListener(): void;
+        };
+    }
+}
+const tokens = [
```
Comment (mizdler) [inline apps/marginfi-v2-ui/src/pages/bridge.tsx:101]: Please take a look at this @edgarpavlovsky

File: apps/marginfi-v2-ui/src/pages/bridge.tsx [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    '0x0000000000000000000000000000000000000000', // SOL
```
Comment (edgarpavlovsky) [inline apps/marginfi-v2-ui/src/pages/bridge.tsx:102]:
```suggestion
    'So11111111111111111111111111111111111111112', // SOL
```

File: apps/marginfi-v2-ui/src/pages/bridge.tsx [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    'EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v', // USDC
+    '7dHbWXmci3dT8UFYWYZweBLXgycu7Y3iL6trKn1Y7ARj', // stSOL
+    'Es9vMFrzaCERmJfrF4H2FYD4KCoNkY11McCe8BenwNYB', // USDT
+    'bSo13r4TkiE4KumL71LsHTPpL2euBYLFx6h9HP3piy1', // bSOL
+    '7vfCXTUXx5WJV5JADk17DUJ4ksgau7utNKj4b963voxs', // ETH
+    '3NZ9JMVBmGAqocybic2c7LQCJScmgsAZ6vQqTDzcqmJh', // BTC
+    'DezXAZ8z7PnrnRJjz3wXBoRgixCa6xjnB7YaB1pPB263', // BONK
+    'DUSTawucrTsGU8hcqRdHDCbuYhCPADMLM2VcCb8VnFnQ', // DUST
+    'kinXdEcpDQeHPEuQnqmUgtYykqKGVFq6CeVX5iAHJq6', // KIN
+    '3NZ9JMVBmGAqocybic2c7LQCJScmgsAZ6vQqTDzcqmJh', //BTC
```
Comment (edgarpavlovsky) [inline apps/marginfi-v2-ui/src/pages/bridge.tsx:115]:
```suggestion
```

File: apps/marginfi-v2-ui/src/pages/bridge.tsx [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+            removeSwapInitiateListener(): void;
+            setSwapCompleteListener(callback: (swap: MayanSwapInfo) => void): void;
+            removeSwapCompleteListener(): void;
+            setSwapRefundListener(callback: (swap: MayanSwapInfo) => void): void;
+            removeSwapRefundListener(): void;
+        };
+    }
+}
+const tokens = [
+    'So11111111111111111111111111111111111111112', // SOL
```
Comment (edgarpavlovsky) [inline apps/marginfi-v2-ui/src/pages/bridge.tsx:102]:
```suggestion
    '0x0000000000000000000000000000000000000000', // SOL
```

**Pull request merged**